src/ca/wolfram/beta/symath/MathConstantNode.java

```diff
@@ -0,0 +1,48 @@
+package ca.wolfram.beta.symath;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Allan Wang on 2017-05-06.
+ */
+public class MathConstantNode implements MathNode {
+
+    private final String variable;
+    private final Map<String, Double> variableMap;
+
+    private MathConstantNode(String variable, Map<String, Double> variableMap) {
+        this.variable = variable;
+        this.variableMap = variableMap;
+    }
+
+    static MathConstantNode create(String variable, Map<String, Double> variableMap) {
+        return new MathConstantNode(variable,variableMap);
+    }
+
+    @Override
+    public NodeType getType() {
+        return NodeType.MATHCONSTANT;
+    }
+
+    @Override
+    public List<MathNode> getChildren() {
+        return null;
+    }
+
+    @Override
+    public boolean isConstant() {
+        return true;
+    }
+
+    @Override
+    public double eval() {
+        return variableMap.containsKey(variable) ? variableMap.get(variable) : 0;
```

Comment:
I think since it is a math constant, it might be better to do a look up of some preprogrammed values. or we can just make sure it is within our variableMap